src/vs/editor/common/controller/cursorDeleteOperations.ts

```diff
@@ -142,62 +143,70 @@ export class DeleteOperations {
-		for (let i = 0, len = selections.length; i < len; i++) {
-			const selection = selections[i];
+			const position = selection.getPosition();
 
-			let deleteSelection: Range = selection;
+			// Unintend when using tab stops and cursor is within indentation
+			if (config.useTabStops && position.column > 1) {
+				const lineContent = model.getLineContent(position.lineNumber);
 
-			if (deleteSelection.isEmpty()) {
-				let position = selection.getPosition();
+				const firstNonWhitespaceIndex = strings.firstNonWhitespaceIndex(lineContent);
+				const lastIndentationColumn = (
+					firstNonWhitespaceIndex === -1
+						? /* entire string is whitespace */ lineContent.length + 1
+						: firstNonWhitespaceIndex + 1
+				);
 
-				if (config.useTabStops && position.column > 1) {
-					let lineContent = model.getLineContent(position.lineNumber);
+				if (position.column <= lastIndentationColumn) {
+					const fromVisibleColumn = CursorColumns.visibleColumnFromColumn2(config, model, position);
+					const toVisibleColumn = CursorColumns.prevIndentTabStop(fromVisibleColumn, config.indentSize);
+					const toColumn = CursorColumns.columnFromVisibleColumn2(config, model, position.lineNumber, toVisibleColumn);
+					return new Range(position.lineNumber, toColumn, position.lineNumber, position.column);
+				}
+			}
 
-					let firstNonWhitespaceIndex = strings.firstNonWhitespaceIndex(lineContent);
-					let lastIndentationColumn = (
-						firstNonWhitespaceIndex === -1
-							? /* entire string is whitespace */lineContent.length + 1
-							: firstNonWhitespaceIndex + 1
-					);
+			return Range.fromPositions(decreasePositionInModelBy1Column(position, model) || position, position);
+		}
 
-					if (position.column <= lastIndentationColumn) {
-						let fromVisibleColumn = CursorColumns.visibleColumnFromColumn2(config, model, position);
-						let toVisibleColumn = CursorColumns.prevIndentTabStop(fromVisibleColumn, config.indentSize);
-						let toColumn = CursorColumns.columnFromVisibleColumn2(config, model, position.lineNumber, toVisibleColumn);
-						deleteSelection = new Range(position.lineNumber, toColumn, position.lineNumber, position.column);
-					} else {
-						deleteSelection = new Range(position.lineNumber, position.column - 1, position.lineNumber, position.column);
-					}
-				} else {
-					let leftOfPosition = MoveOperations.left(config, model, position.lineNumber, position.column);
-					deleteSelection = new Range(
-						leftOfPosition.lineNumber,
-						leftOfPosition.column,
-						position.lineNumber,
-						position.column
-					);
-				}
+		function decreasePositionInModelBy1Column(position: Position, model: ICursorSimpleModel): Position | undefined {
+			if (position.column > 1) {
+				return position.delta(0, -1);
+			} else if (position.lineNumber > 1) {
+				const newLine = position.lineNumber - 1;
+				return new Position(newLine, model.getLineMaxColumn(newLine));
+			} else {
+				return undefined;
```

Comment:
this could just return `position` and always be defined, which would allow the `|| position` to be removed in the calling site.

Reply:
I considered that, but that would violate the semantics of `decreasePositionInModelBy1Column` to return a position that is actually decreased.
I could rename the method to `decreasePositionInModelBy1ColumnOrDefault` and make it always defined?

Reply:
I don't have strong feelings either way.